engines/terraform/resolve.go

```
@@ -88,7 +111,11 @@ func (td *TerraformDeployment) resolveTokensForModule(intentName string, resourc
 	for property, value := range resource.Properties {
 		resolvedValue, err := td.resolveValue(intentName, value)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to resolve property '%s' for intent '%s': %w", property, intentName, err)
+		}
+		// Skip nil values and empty strings
+		if isUnset(resolvedValue) {
+			continue
```

What are the circumstances that are leading the nil references? Is there something we can do to handle the root cause, rather than silently swallowing what could be a configuration error?

---

```
Suga Build Panic Report
========================
Timestamp: 2025-09-19T10:05:23-06:00
Target: suga/rkplatform@19
Panic: parameter value is required, but nil was provided

Stack Trace:
goroutine 1 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go/1.24.4/libexec/src/runtime/debug/stack.go:26 +0x64
github.com/nitrictech/suga/engines/terraform.(*TerraformEngine).Apply.func1()
	/Users/rs/DevelopmentV2/suga/engines/terraform/terraform.go:95 +0x44
panic({0x103b946c0?, 0x1400094e010?})
	/opt/homebrew/Cellar/go/1.24.4/libexec/src/runtime/panic.go:792 +0x124
github.com/hashicorp/terraform-cdk-go/cdktf.(*jsiiProxy_TerraformHclModule).Set(0x14000660678, 0x1400094e000, {0x0, 0x0})
	/Users/rs/go/pkg/mod/github.com/hashicorp/terraform-cdk-go/cdktf@v0.20.12/TerraformHclModule.go:461 +0x90
github.com/nitrictech/suga/engines/terraform.(*TerraformDeployment).resolveTokensForModule(0x14000896600, {0x14000982ce0, 0xd}, 0x1400017fef0?, {0x103f469a8, 0x14000660678})
	/Users/rs/DevelopmentV2/suga/engines/terraform/resolve.go:93 +0x80
github.com/nitrictech/suga/engines/terraform.(*TerraformEngine).Apply(0x140003162d0, 0x140006945a0)
	/Users/rs/DevelopmentV2/suga/engines/terraform/terraform.go:162 +0x928
github.com/nitrictech/suga/cli/internal/build.(*BuilderService).BuildProjectForTarget(0x140005ed680, 0x140006945a0, {0x14000998d98, 0x12})
	/Users/rs/DevelopmentV2/suga/cli/internal/build/build.go:53 +0x300
github.com/nitrictech/suga/cli/pkg/app.(*SugaApp).Build(0x140005f2708)
	/Users/rs/DevelopmentV2/suga/cli/pkg/app/suga.go:481 +0x1fc
github.com/nitrictech/suga/cli/cmd.NewBuildCmd.func1(0x140006a4e00?, {0x103416fca?, 0x4?, 0x103416fce?})
	/Users/rs/DevelopmentV2/suga/cli/cmd/nitric.go:79 +0x48
github.com/spf13/cobra.(*Command).execute(0x140006e1808, {0x1050fba40, 0x0, 0x0})
	/Users/rs/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0x82c
github.com/spf13/cobra.(*Command).ExecuteC(0x140006e0008)
	/Users/rs/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x384
github.com/spf13/cobra.(*Command).Execute(0x103f44400?)
	/Users/rs/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071 +0x1c
main.main()
	/Users/rs/DevelopmentV2/suga/cli/main.go:40 +0x138
```

---

```yaml
{
  "name": "rkplatform",
  "libraries": { "Official": "suga/Official@v0.0.1" },
  "variables":
    {
      "neon_project_id":
        {
          "type": "string",
          "description": "",
          "default": null,
          "nullable": false,
        },
    },
  "services":
    {
      "aws-fargate":
        {
          "plugin": "Official/aws-fargate",
          "properties":
            {
              "alb_arn": "${infra.aws-loadbalancer.arn}",
              "alb_security_group": "", # this shouldn't be set at all if the value has been cleared.
              "container_port": "8080",
              "security_groups": "",
              "subnets": "",
              "vpc_id": "",
            },
  ...

---

this should be fixed by: https://github.com/nitrictech/platform/pull/223